Formula/dspdfviewer.rb

```diff
@@ -27,65 +27,32 @@ class Dspdfviewer < Formula
   depends_on "libpng"
   depends_on "libtiff"
   depends_on "openjpeg"
+  depends_on "poppler-qt5"
   depends_on "qt@5"
 
-  resource "poppler" do
-    url "https://poppler.freedesktop.org/poppler-0.65.0.tar.xz"
-    sha256 "89c8cf73f83efda78c5a9bd37c28f4593ad0e8a51556dbe39ed81e1ae2dd8f07"
+  on_linux do
+    depends_on "gcc"
   end
 
-  resource "font-data" do
-    url "https://poppler.freedesktop.org/poppler-data-0.4.9.tar.gz"
-    sha256 "1f9c7e7de9ecd0db6ab287349e31bf815ca108a5a175cf906a90163bdbe32012"
-  end
+  fails_with gcc: "5"
 
   def install
-    ENV.cxx11
-
-    resource("poppler").stage do
-      system "cmake", ".", *std_cmake_args,
-                           "-DCMAKE_INSTALL_PREFIX=#{libexec}",
-                           "-DBUILD_GTK_TESTS=OFF",
-                           "-DENABLE_CMS=none",
-                           "-DENABLE_GLIB=ON",
-                           "-DENABLE_QT5=ON",
-                           "-DWITH_GObjectIntrospection=ON",
-                           "-DENABLE_XPDF_HEADERS=ON"
-      system "make", "install"
-
-      libpoppler = (libexec/"lib/libpoppler.dylib").readlink
-      to_fix = ["#{libexec}/lib/libpoppler-cpp.dylib", "#{libexec}/lib/libpoppler-glib.dylib",
-                "#{libexec}/lib/libpoppler-qt5.dylib", *Dir["#{libexec}/bin/*"]]
-
-      to_fix.each do |f|
-        macho = MachO.open(f)
-        macho.change_dylib("@rpath/#{libpoppler}", "#{libexec}/lib/#{libpoppler}")
-        macho.write!
-      end
-
-      resource("font-data").stage do
-        system "make", "install", "prefix=#{libexec}"
-      end
-    end
-
-    ENV.prepend_path "PKG_CONFIG_PATH", "#{libexec}/lib/pkgconfig"
-    ENV.prepend "LDFLAGS", "-L#{libexec}/lib"
-
     mkdir "build" do
       system "cmake", "..", *std_cmake_args,
                             "-DRunDualScreenTests=OFF",
                             "-DUsePrerenderedPDF=ON",
                             "-DUseQtFive=ON"
       system "make", "install"
     end
-
-    libpoppler = (libexec/"lib/libpoppler-qt5.dylib").readlink
-    macho = MachO.open(bin/"dspdfviewer")
-    macho.change_dylib("@rpath/#{libpoppler}", "#{libexec}/lib/#{libpoppler}")
-    macho.write!
   end
 
   test do
-    system bin/"dspdfviewer", "--help"
+    on_macos do
+      system "#{bin}/dspdfviewer", "--help"
+    end
+    on_linux do
+      assert_match("qt.qpa.xcb: could not connect to display",
+                   shell_output("#{bin}/dspdfviewer --help 2>&1", 134))
+    end
```

Comment:
after digging, it looks like the linux error is due to our docker container lacking X11? https://github.com/therecipe/qt/issues/775#issuecomment-456899597

not sure if there's a more direct solution, but this is one workaround: 
```suggestion
    ENV["QT_QPA_PLATFORM"] = "minimal" if OS.linux?
    system bin/"dspdfviewer", "--help"
```